test/parallel/test-async-local-storage-exit-does-not-leak.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const common = require('../common');
+const { AsyncLocalStorage } = require('async_hooks');
+
+const als = new AsyncLocalStorage();
+
+// The als instance should be getting removed from the storageList in
+// lib/async_hooks.js when exit(...) is called, therefore when the nested runs
+// are called there should be no copy of the als in the storageList to run the
+// _propagate method on.
+als._propagate = common.mustNotCall(als._propagate);
```

Comment:
The optional argument for `mustNotCall()` is a message string.

```suggestion
als._propagate = common.mustNotCall();
```